### DISCUSSION TITLE
How to use Semantic Kernel plugins inside this repo ?

### DISCUSSION THREAD

**Discussion by william-daconceicao** [microsoft/chat-copilot#999] [General]
Hello everyone ! I' trying to use Semantic Kernel plugin but it only allow OpenAI plugin inside the appm do you have any clue to use it? 
I want to use this kind of plugins
https://github.com/microsoft/semantic-kernel/tree/main/prompt_template_samples/CalendarPlugin/AssistantShowCalendarEvents
Instead of this kind
https://github.com/SamilMehdiyev/openai-weather-plugin-golang/blob/main/.well-known/ai-plugin.json

Thanks a lot

**Comment by aaronba** [comment]
Hey @william-daconceicao , I believe that SK CC can use NativePlugins and SemanticPlugins 
This is documented [Here](https://github.com/microsoft/chat-copilot/tree/main/webapi#optional-custom-semantic-kernel-setup)

I have successfully called NativePlugins by uncommenting the setting in appsettings.json and modifying the code to call the Custom hook in Extensions/SemanticKernelExtensions.cs

// Uncomment the following line and pass in your custom hook.
   builder.Services.AddKernelSetupHook(RegisterPluginsAsync);
   
  Once uncommented, it should call the code to load the NativePlugin or SemanticPlugin plugins in the directory.
